db/compaction/compaction_job.cc

```diff
@@ -280,13 +284,15 @@ void CompactionJob::Prepare() {
     RecordInHistogram(stats_, NUM_SUBCOMPACTIONS_SCHEDULED,
                       compact_->sub_compact_states.size());
   } else {
-    compact_->sub_compact_states.emplace_back(c, std::nullopt, std::nullopt,
+    auto single_subcompact = known_single_subcompact.value_or(
+        std::make_pair(std::optional<Slice>{}, std::optional<Slice>{}));
```

Comment:
Question for my own learning. I believe that `std::optional<Slice>{}` would also return false for `has_value()`. This just explicitly shows the intention of creating an empty optional Slice. Please correct me if I am wrong.

Reply:
I guess the code is a little too clever in dealing with the nested optionals etc. I'll rewrite it.